datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateDescriptionResolver.java

```diff
@@ -107,6 +115,36 @@ private CompletableFuture<Boolean> updateDomainDescription(Urn targetUrn, Descri
     });
   }
 
+  // If updating schema field description fails, try again on a sibling until there are no more siblings to try. Then throw if necessary.
+  private Boolean attemptUpdateDatasetDescription(
+      @Nonnull final Urn targetUrn,
+      @Nonnull final DescriptionUpdateInput input,
+      @Nonnull final QueryContext context,
+      @Nonnull final HashSet<Urn> attemptedUrns,
+      @Nonnull final List<Urn> siblingUrns
+  ) {
+    attemptedUrns.add(targetUrn);
+    try {
+      DescriptionUtils.validateFieldDescriptionInput(targetUrn, input.getSubResource(), input.getSubResourceType(),
+          _entityService);
+
+      final Urn actor = CorpuserUrn.createFromString(context.getActorUrn());
+      DescriptionUtils.updateFieldDescription(input.getDescription(), targetUrn, input.getSubResource(), actor,
+          _entityService);
+      return true;
+    } catch (Exception e) {
+      final Optional<Urn> siblingUrn = SiblingsUtils.getNextSiblingUrn(siblingUrns, attemptedUrns);
+
+      if (siblingUrn.isPresent()) {
+        log.warn("Failed to update description for input {}, trying sibling urn {} now.", input.toString(), siblingUrn.get());
+        return attemptUpdateDatasetDescription(siblingUrn.get(), input, context, attemptedUrns, siblingUrns);
+      } else {
+        log.error("Failed to perform update against input {}, {}", input.toString(), e.getMessage());
+        throw new RuntimeException(String.format("Failed to perform update against input %s", input.toString()), e);
+      }
+    }
+  }
+
   private CompletableFuture<Boolean> updateDatasetDescription(Urn targetUrn, DescriptionUpdateInput input, QueryContext context) {
```

Comment:
Is this method named correctly? It seems to only support Dataset Field description updates. I'm not sure why historically this would be the case but maybe it's time to consider renaming it

Reply:
yup you're definitely right and I remember discovering this some time back and being confused lol i'll update this to be `updateDatasetSchemaFieldDescription` to at least make that more obvious